deps/Eigen/Eigen.cmake

```diff
@@ -0,0 +1,8 @@
+orcaslicer_add_cmake_project(
```

Comment:
Why?
Orca has Eigen in the source: src\eigen\Eigen

Reply:
Will test with just the find portion and not the additional dep, will update shortly.

Reply:
It does look like this can be a more simple change, will update the PR.

Reply:
Can you double check? Might be an env issue?
It compiles fine in CI/CD

Reply:
I've double checked and the only addition that is needed is what's in the current PR, a simple find addition: https://github.com/SoftFever/OrcaSlicer/pull/5202/commits/ce91025241e5cf321558d0828d6e49a78209ae9e

Reply:
I'm afraid this is not right.
By adding this, you are introducing two Eigen versions into Orca, once is within Orca's source code, one is from your OS.
You can try to uninstall your system's Eigen and rebuild deps.

Reply:
I don't have one installed on my OS.  Only the one in the source tree.

Reply:
I've dug into this more.  If I have an eigen package installed when the deps are built, then OpenCV's `OpenCVModules.cmake` has this line:

```
INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:dl>;\$<LINK_ONLY:m>;\$<LINK_ONLY:pthread>;\$<LINK_ONLY:rt>;\$<LINK_ONLY:Eigen3::Eigen>;\$<LINK_ONLY:libjpeg-turbo>;\$<LINK_ONLY:libpng>;/home/jamin/proj/3d-printing/OrcaSlicer/deps/build/destdir/usr/local/lib/libtiff.a;/lib64/libz.so;/lib64/libz.so;\$<LINK_ONLY:va>;\$<LINK_ONLY:va-drm>"
```

If I remove the eigen package **and** rebuild the deps, the `OpenCVModules.cmake` has this line:

```
INTERFACE_LINK_LIBRARIES "\$<LINK_ONLY:dl>;\$<LINK_ONLY:m>;\$<LINK_ONLY:pthread>;\$<LINK_ONLY:rt>;\$<LINK_ONLY:libjpeg-turbo>;\$<LINK_ONLY:libpng>;/home/jamin/proj/3d-printing/OrcaSlicer/deps/build/destdir/usr/local/lib/libtiff.a;/lib64/libz.so;/lib64/libz.so;\$<LINK_ONLY:va>;\$<LINK_ONLY:va-drm>"
```

The difference between the two being:

```
\$<LINK_ONLY:Eigen3::Eigen>;
```

This means that the OpenCV dep is changing its compile behavior based on which system packages are installed.

That doesn't seem desirable to me.  IMO, it's one thing to indicate which packages a user needs to have installed in order to build software.  It's another thing entirely to tell a user they _must not_ have a package installed in order to build.

While the `OpenCV.cmake` has this line:

```
       -DWITH_EIGEN=ON
```

I don't believe this is actually causing OpenCV to be built with Orca's in-tree Eigen (src/eigen), but instead simply triggering the above edge case.

I've built deps and Orca with the above set to `OFF` and so far see no negative effects.

Reply:
This has also been independently hit by another member on the discord.
https://discord.com/channels/1137181739773603922/1137187844696318023/1235691546943295568

Reply:
Thanks, Collins.
What Linux distribution are you using?
I can take a look when I'm free.
However the proposed change here isn't desirable.
It might seem harmless at first, but trust me, it'll lead to trouble eventually.
C++ has many quirks unfortunately

Reply:
Arch Linux.